src/main.c

```diff
@@ -3261,15 +3261,20 @@ source_startup_scripts(mparm_T *parmp)
 	/*
 	 * Try to read initialization commands from the following places:
 	 * - environment variable VIMINIT
+	 * - defaults.vim unless $VIM_NO_SOURCE_DEFAULTS is set
 	 * - user vimrc file (s:.vimrc for Amiga, ~/.vimrc otherwise)
 	 * - second user vimrc file ($VIM/.vimrc for Dos)
 	 * - environment variable EXINIT
 	 * - user exrc file (s:.exrc for Amiga, ~/.exrc otherwise)
 	 * - second user exrc file ($VIM/.exrc for Dos)
 	 * The first that exists is used, the rest is ignored.
 	 */
+	int	loaded_defaults = NOTDONE;
 	if (process_env((char_u *)"VIMINIT", TRUE) != OK)
 	{
+	    if (mch_getenv("VIM_NO_SOURCE_DEFAULTS") == NULL)
```

Comment:
If it's going to be unconditionally loaded, it seems like it should happen before the system vimrc so the typical "defaults can be overridden by system config which can be overridden by user config" pattern is followed.